TCD1304.py

```diff
@@ -62,29 +58,32 @@ def power_source ():        # puts a voltage of 4V on PV1
         ps.pv1 = 4
         ps.pv1
     
+    def analog_signal_read (self, block:False):
+        self.scope = Oscilloscope()
+        self.scope.select_range('CH1', max_voltage_output)   # voltagerange should be fitted to the sensors output for better resolution. sensor otput is between 2V-3V due to datasheet
+        self.scope.configure_trigger(channel = 'CH1', voltage = min_voltage_output ) # starts recording , when voltage is over a certain level
+        self.scope.capture(channels=1, samples=integration_elements*samples_per_element, timegap=integration_time/samples_per_element)
+        y1, y2 = self.scope.fetch_data()

```

Comment:
You will need to run the experiment before fetching the samples. Calling `capture` in non-blocking mode and then immediately calling `fetch_data` is functionally identical to just using blocking mode.

The call order should be:
1. `scope.capture(..., block=False)  # Prime the oscilloscope for capture.`
2. `icg_clock()  # Start the sensor output, triggering the oscilloscope.`
3. `x, y = scope.fetch_data()  # Fetch collected samples after everything is done.`